pkg/client/delegation/client_integration_test.go

```diff
@@ -0,0 +1,190 @@
+//go:build integration
+
+package delegation_test
+
+// TODO(@h5law): Figure out how to use real components of the localnet
+//	- Create app and gateway actors
+//  - Stake them
+//  - Delegate to the gateway
+//  - Undelegate from the gateway
+// Currently this test doesn't work, because (I think) it is using a mock
+// keeper etc and this isnt actually interacting with the localnet where
+// the DelegationClient is listening for events from.
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	keepertest "github.com/pokt-network/poktroll/testutil/keeper"
```

Comment:
One of the reasons I believe you're struggling with this test is because we're trying to use `keeper` outside of on-chain code. Seems to be the first instance of it:

<img width="507" alt="Screenshot 2023-12-04 at 7 06 48 PM" src="https://github.com/pokt-network/poktroll/assets/1892194/de4e8e48-e954-4764-8bd6-4286c24eeffb">

If you look in `x/supplier/client/cli/query_claim_test.go`, what we do is:
1. Create an in-memory network
2. Use the test CLI to create claim objects

@bryanchriswhite is reworking that code in #236 but the idea is the same and what I think you want instead of using the keepers directly.

Reply:
@Olshansk https://arc.net/l/quote/npxdzlgu this approach was in progress and I will go back to it after your comment

Reply:
Tried and failed left comment, think its something to do with the networks subscribiton to events unsure atm.

Reply:
@h5law Wanted to ask to leave these sort of threads open in the future or let the author of the thread resolve it. Even if we don't necessarily tackle the work in this PR, it helps keep track of followup work.